### PR TITLE
util/log: fix expected log file name

### LINKED ISSUE #9823: github.com/cockroachdb/cockroach/util/log: TestGetLogReader failed under stress
SHA: https://github.com/cockroachdb/cockroach/commits/d54fe4d3fad3c1b2ecc3ff28f5fac5066405125d

Stress build found a failed test:

```
```

### PR DESCRIPTION
**Pull request by tamird**
Fixes #9823.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9890)

### CODE REVIEW

File: util/log/file.go [hunk 1]
```diff
@@ -266,13 +264,6 @@ func ListLogFiles() ([]FileInfo, error) {
 // feature that relative paths will be searched in both the current
 // directory and this process's log directory.
 func GetLogReader(filename string, restricted bool) (io.ReadCloser, error) {
-	if !restricted {
```
Comment (tschottdorf) [inline util/log/file.go]: what's this?

Comment (tamird) [inline util/log/file.go]: This code appears to be garbage that does nothing.

**Comment by tschottdorf** [comment]
LGTM

**Comment by RaduBerinde** [comment]
LGTM

**Pull request merged**